### PR TITLE
ROX-24631: Make ocp-nongroovy test work

### PR DESCRIPTION
**Pull request by porridge**
## Description

This PR:
- improves the run-e2e-tests.sh script to be more robust
  - Tolerate unset env var in export_central_basic_auth_creds.
  - Add a hint to README for using another launcher.
  - Update slack channel name in README.
  - Set CWD before invoking make tag.
  - Set default GOPATH if unset.
  - Do not invoke kubectl before setting up gcp access.
  - Collect our CRs too.
- temporarily skips a few nongroovy tests/setups on ocp. This way we will be able to start running this job on OCP by default soon, and keep iterating on it and fix these cases one by one, while preventing new regressions from being added.
  - Be more verbose when checking port availability.
  - Temporarily skip endpoints test and its prep on OCP: ROX-24688.
  - Temporarily skip a case in TestCASetup on OCP: ROX-24688.
  - Temporarily skip TestFeatureFlagSettings on ocp: ROX-24688.
  - Temporarily skip TestGraphQLSorting on OCP: ROX-24688.
  - Temporarily skip TestMetadataIsSetCorrectly on OCP: ROX-24688.
  - Temporarily skip failing proxy tests on OCP: ROX-24688.
  - Temporarily skip TestExcludedScopes and TestBackup on OCP as their pods crash: ROX-24688.
  - Temporarily skip remaining tests on OCP to save time: ROX-24688.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] ~Evaluated and added CHANGELOG entry if required~
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- [ ] Run gke and ocp job in CI 10 times
- [x] Make sure the skips do happen, and only on OCP

GKE:
Screenshot from 2024-06-12 16-23-08

OCP:
Screenshot from 2024-06-12 16-23-52

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]
I'd like to see that `nit:` changed throughout.

File: tests/e2e/run.sh [hunk 1]
```diff
@@ -58,32 +58,49 @@ test_e2e() {
     store_test_results "tests/all-tests-results" "all-tests-results"
     [[ ! -f FAIL ]] || die "e2e API tests failed"
 
-    # Give some time for previous tests to finish up
-    wait_for_api
+    if [[ ${ORCHESTRATOR_FLAVOR:-} = openshift ]]; then
```
[the remark this answers is not in the document]
Comment (porridge) [reply tests/e2e/run.sh]: Done, PTAL

**Review by gavin-stackrox — APPROVED** [review]

**Comment by gavin-stackrox** [comment]
CI looks good. Will merge.

**Pull request merged**